lib/pages/company_site/eti/table_products.rb

```diff
@@ -0,0 +1,318 @@
+module CompanySite
+  module ETI
+    class Table
+      class Products < self
+        include CompanySite::ETI
+
+        EXISTS = {
+          available: 'В наличии',
+          not_available: 'Нет в наличии',
+          order: 'Под заказ',
+          awaiting: 'Ожидается поступление',
+          none: 'Не указано'
+        }.freeze
+
+        button(:add_new_product, css: '.new.js-add-product')
+        elements(:products, :row, css: '*[id^="product-item"]')
+        elements(:names, :cell, css: '.js-eti-name')
+        elements(:rubrics, :cell, css: '.js-eti-rubric')
+        elements(:images, :cell, css: '*[id^="product-item"] *[class*="ibb"]')
+        elements(:images_counters, :div, css: '*[id^="product-item"] *[class="ip-count"]')
+        elements(:public_state_icon, :cell, css: '.js-eti-status')
+        elements(:battery_level, :cell, css: '.js-battery-wrapper')
+        elements(:battery_title, :cell, css: '.battery')
+        elements(:rubric_link, :cell, css: '.js-rubric-preview-link')
+        elements(:exists_link, :cell, css: '.js-eti-exists a')
+        elements(:exists_values, :link, css: '.ui-autocomplete li a')
+        elements(:announces, :cell, css: '.js-eti-announce')
+        elements(:prices, :cell, css: '.js-eti-price [title="Изменить цену"]')
+        elements(:wholesale_prices, :cell, css: '.js-eti-wholesaleprice [title="Указать оптовую цену"]')
+        elements(:traits_link, :cell, css: '.js-eti-traits > div')
+        elements(:descriptions, :cell, css: '.js-eti-description')
+        elements(:groups, :cell, css: '.js-group-preview-link')
+
+        div(:edit_box, css: '.editbox')
+
+        button(:delete_product_icon, css: '.action-product .js-delete-product')
+        elements(:copy_product_icons, css: '.js-copy-product')
+
+        def product(param = {})
+          if param.key? :name
+            name_element = names_elements.select { |n| n.text == param[:name] }.first
+            products_elements[names_elements.index(name_element)]
+          elsif param.key? :index
+            products_elements[param[:index]]
+          else
+            nil
+          end
+        end
+
+        def add_product(params = {})
```

Comment:
Assignment Branch Condition size for add_product is too high. [53.76/35]<br>Cyclomatic complexity for add_product is too high. [10/6]<br>Perceived complexity for add_product is too high. [10/7]